__tests__/pet.test.js

```diff
@@ -0,0 +1,149 @@
+	it('sets initial fitness to 10', () => {
+		expect(pet.fitness).toEqual(10);
+	});
+});
+
+describe('growUp', () => {
+	it('increments the pet age by one', () => {
+		pet.growUp();
+		expect(pet.age).toEqual(1);
+	});
+	it('increases the pet hunger by 5', () => {
+		pet.growUp();
+		expect(pet.hunger).toEqual(5);
+	});
+	it('decreases the pet fitness by 3', () => {
+		pet.growUp();
+		expect(pet.fitness).toEqual(7);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.growUp()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('walk', () => {
+	it('increases fitness by 4', () => {
+		pet.fitness = 5;
+		pet.walk();
+		expect(pet.fitness).toEqual(9);
+	});
+	it('stops fitness going over 10', () => {
+		pet.fitness = 9;
+		pet.walk();
+		expect(pet.fitness).toEqual(10);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.walk()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('feed', () => {
+	it('decreases pet hunger level by 3', () => {
+		pet.hunger = 5;
+		pet.feed();
+		expect(pet.hunger).toEqual(2);
+	});
+	it('stops hunger going less than 0', () => {
+		pet.hunger = 1;
+		pet.feed();
+		expect(pet.hunger).toEqual(0);
+	});
+	it('throws an error if the pet is not alive', () => {
+		pet.age = 30;
+		expect(() => pet.feed()).toThrow('Your pet is no longer alive :(');
+	});
+});
+
+describe('checkUp', () => {
+	it('return "I need a walk" if fitness less than 4 and hunger less than 5', () => {
+		pet.hunger = 1;
+		pet.fitness = 1;
+		expect(pet.checkUp()).toEqual('I need a walk');
+	});
+	it('return "I am hungry" if hunger greater than 4 and fitness greater than 3', () => {
+		pet.hunger = 5;
+		pet.fitness = 9;
+		expect(pet.checkUp()).toEqual('I am hungry');
+	});
+	it('return "I am hungry AND I need a walk" if hunger greater than 4 and fitness less than 4', () => {
+		pet.hunger = 9;
+		pet.fitness = 1;
+		expect(pet.checkUp()).toEqual('I am hungry AND I need a walk');
+	});
+
+	it('return "I feel great!" if hunger less than 5 and fitness greater than 3', () => {
+		expect(pet.checkUp()).toEqual('I feel great!');
+	});
+	it('returns message if the pet is not alive', () => {
+		pet.age = 30;
+		expect(pet.checkUp()).toEqual('Your pet is no longer alive :(');
+	});
+});
+
+describe('isAlive', () => {
+	it('returns false if fitness < 1', () => {
+		pet.fitness = 0;
+		expect(pet.isAlive).toEqual(false);
+	});
+	it('returns true if fitness > 0', () => {
+		expect(pet.isAlive).toEqual(true);
+	});
+	it('returns false if hunger > 9', () => {
+		pet.hunger = 10;
+		expect(pet.isAlive).toEqual(false);
+	});
+	it('returns true if hunger < 10', () => {
+		expect(pet.isAlive).toEqual(true);
+	});
+	it('returns false if age > 29', () => {
+		pet.age = 30;
+		expect(pet.isAlive).toEqual(false);
+	});
+	it('returns true if age < 30', () => {
+		expect(pet.isAlive).toEqual(true);
+	});
+});
+
+describe('adopt a child', () => {
```

Comment:
Due to these methods working differently (IE one accepting a string and the other expecting an instance of `Pet`) these tests are not right. (Disclaimer: if the intention is for adopted children to be stored as strings and non adopted to be `Pet` objects, an explanation in a comment would be very helpful as I have no way of knowing what the intention was)
There should be no logic needed inside the test that is doing the same job as the object being tested. 

In the first test, you are creating an instance of `Pet` then passing it into the `adoptChild()` method, then checking for the type of the whole array. This doesnt make sense, as if you were to remove lines 130 and 131 and run the tests they would still pass! This is because an empty array is actually an object. A more worthwhile test would be 'check adopted child is instance of Pet', and checking if the first value in that array is an instance of Pet as expected:
```
expect(pet.children[0]).toBeInstanceOf(Pet);
```